mm2src/kdf_walletconnect/src/chain.rs

```diff
@@ -0,0 +1,136 @@
+use mm2_err_handle::prelude::{MmError, MmResult};
+use relay_rpc::rpc::params::session::{ProposeNamespace, ProposeNamespaces};
+use serde::{Deserialize, Serialize};
+use std::{collections::{BTreeMap, BTreeSet},
+          str::FromStr};
+
+use crate::error::WalletConnectError;
+
+pub(crate) const SUPPORTED_PROTOCOL: &str = "irn";
+
+pub(crate) const COSMOS_SUPPORTED_METHODS: &[&str] = &["cosmos_getAccounts", "cosmos_signDirect", "cosmos_signAmino"];
+pub(crate) const COSMOS_SUPPORTED_CHAINS: &[&str] = &["cosmos:cosmoshub-4"];
```

Comment:
Instead of hardcoding these, what about adding a flag in coins file like `wallet_connect: bool`?

Reply:
these are for default initialization only.. when creating a WalletConnect session use can provide chains they want to interact with. E.g https://github.com/KomodoPlatform/komodo-defi-framework/pull/2223#issue-2529262185